tests/unit/DynamicSegments/RequirementsCheckerTest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace MailPoet\DynamicSegments;
+
+use MailPoet\DynamicSegments\Filters\EmailAction;
+use MailPoet\DynamicSegments\Filters\WooCommerceCategory;
+use MailPoet\Models\DynamicSegment;
+use MailPoet\WooCommerce\Helper;
+
+class RequirementsCheckerTest extends \MailPoetUnitTest {
+
+  /** @var Helper|\PHPUnit_Framework_MockObject_MockObject */
+  private $woocommerce_helper;
+
+  /** @var RequirementsChecker */
+  private $requirements_checker;
+
+  function _before() {
+    parent::_before();
+    if (!defined('MP_SEGMENTS_TABLE')) define('MP_SEGMENTS_TABLE', '');
+    $this->woocommerce_helper = $this
+      ->getMockBuilder(Helper::class)
+      ->setMethods(['isWooCommerceActive'])
+      ->getMock();
+    $this->requirements_checker = new RequirementsChecker($this->woocommerce_helper);
+  }
+
+  function testShouldntBlockSegmentIfWooCommerceIsActive() {
+    $this->woocommerce_helper->method('isWooCommerceActive')->willReturn(true);
+    $segment = DynamicSegment::create();
+    $segment->setFilters([new WooCommerceCategory(1)]);
+    expect($this->requirements_checker->shouldSkipSegment($segment))->false();
+  }
+
+  function testShouldntBlockWooCommerceSegmentIfWooCommerceIsInactive() {
```

Comment:
The name - `testShouldnt` or `testShould`? I think the latter in this case, right?